index.d.ts

```diff
@@ -0,0 +1,35 @@
+import {IncomingMessage} from 'http';
+
+declare type KnownProperties =
+	| 'complete'
+	| 'destroy'
+	| 'headers'
+	| 'httpVersion'
+	| 'httpVersionMinor'
+	| 'httpVersionMajor'
+	| 'method'
+	| 'rawHeaders'
+	| 'rawTrailers'
+	| 'setTimeout'
+	| 'socket'
+	| 'statusCode'
+	| 'statusMessage'
+	| 'trailers'
+	| 'url';
+
+/** Helper type to flatten nested mapped types */
+declare type Identity<T> = {} & {[K in keyof T]: T[K]};
```

Comment:
This converts something like: `Pick<IncomingMessage, Exclude<"complete", keyof T> | Exclude<"destroy", keyof T> | Exclude<"headers", keyof T> | Exclude<"httpVersion", keyof T> | Exclude<"httpVersionMinor", keyof T> | Exclude<"httpVersionMajor", keyof T> | Exclude<"method", keyof T> | Exclude<"rawHeaders", keyof T> | Exclude<"rawTrailers", keyof T> | Exclude<"setTimeout", keyof T> | Exclude<"socket", keyof T> | Exclude<"statusCode", keyof T> | Exclude<"statusMessage", keyof T> | Exclude<"trailers", keyof T> | Exclude<"url", keyof T>>` (which is what `mimicResponse` gets interpreted as) into:
```ts
{
    complete: boolean;
    destroy: (error?: Error | undefined) => void;
    headers: http.IncomingHttpHeaders;
    httpVersion: string;
    httpVersionMinor: number;
    httpVersionMajor: number;
    method?: string | undefined;
    ... 7 more ...;
    url?: string | undefined;
}
```